hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/FSWALEntry.java

```diff
@@ -50,25 +50,40 @@
   // The below data members are denoted 'transient' just to highlight these are not persisted;
   // they are only in memory and held here while passing over the ring buffer.
   private final transient long txid;
+
+  /**
+   * If false, means this is a meta edit written by the hbase system itself. It was not in
+   * memstore. HBase uses these edit types to note in the log operational transitions such
+   * as compactions, flushes, or region open/closes.
+   */
   private final transient boolean inMemstore;
+
+  /**
+   * Set if this is a meta edit and it is of close region type.
+   */
   private final transient boolean closeRegion;
+
   private final transient RegionInfo regionInfo;
   private final transient Set<byte[]> familyNames;
   private final transient ServerCall<?> rpcCall;
 
+  /**
+   * @param inMemstore If true, then this is a data edit, one that came from client. If false, it
+   *   is a meta edit made by the hbase system itself and is for the WAL only.
+   */
   FSWALEntry(final long txid, final WALKeyImpl key, final WALEdit edit, final RegionInfo regionInfo,
-    final boolean inMemstore, boolean closeRegion, ServerCall<?> rpcCall) {
+    final boolean inMemstore, ServerCall<?> rpcCall) {
     super(key, edit);
     this.inMemstore = inMemstore;
-    this.closeRegion = closeRegion;
+    this.closeRegion = !inMemstore && edit.isRegionCloseMarker();
```

Comment:
OK, so we added a flag to WALEdit, then let's also purge the inMemstore flag here?

Reply:
Well, inMemstore is false for all meta/market types. It is consulted when we go to do update on SequenceIdAccounting. If it is false, we don't bother updating 'lowest' sequenceid.

Unless I'm missing something, we need it still.

Reply:
I mean we do not need to pass inMemstore flag in WAL.append, also add an isInMemstore method in WALEdit class, just like what we have done for isRegionCloseMarker here.